MainPage.xaml.cs

```diff
@@ -22,81 +27,67 @@ public sealed partial class MainPage : Page
         public MainPage()
         {
             this.InitializeComponent();
-
-            TextBlock emptyMessageTextBLock = this.FindName("EmptyMsg") as TextBlock;
-            emptyMessageTextBLock.Visibility = Visibility.Collapsed;
-
         }
 
         private async void SearchButton_Click(object sender, RoutedEventArgs e)
         {
             Debug.WriteLine("Sending Request ... ");
 
-            Windows.Web.Http.HttpClient httpClient = new Windows.Web.Http.HttpClient();
-
-            var headers = httpClient.DefaultRequestHeaders;
-            string header = "ie";
-            if (!headers.UserAgent.TryParseAdd(header))
+            string uri = "";
+            if (this.Keyword.Text.Length > 0)
             {
-                throw new Exception("Invalid header value: " + header);
+                uri = "https://api.github.com/search/repositories?q=" + this.Keyword.Text;
             }
```

Comment:
@kimchhunnn if your keyword is empty, your request uri is empty. Does it make sense ?

Reply:
@kenphanith When the keyword is empty, it will go to else case and return void. No further execution.

Reply:
@kimchhunnn It make sense ?

Reply:
@kenphanith May I have your idea, please?